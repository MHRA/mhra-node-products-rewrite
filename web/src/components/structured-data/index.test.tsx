import renderer from 'react-test-renderer';
import React from 'react';
import {
  DrugListStructuredData,
  DrugStructuredData,
  SubstanceListStructuredData,
  SubstanceStructuredData,
} from './index';

describe(SubstanceStructuredData, () => {
  it('should render', () => {
    const component = renderer
      .create(<SubstanceStructuredData substanceName={'My cool substance'} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});

describe(SubstanceListStructuredData, () => {
  it('should render', () => {
    const component = renderer
      .create(
        <SubstanceListStructuredData
          substanceNames={['First cool substance', 'Second cool substance']}
        />,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});

describe(DrugStructuredData, () => {
  it('should render', () => {
    const component = renderer
      .create(<DrugStructuredData drugName={'My cool drug'} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});

describe(DrugListStructuredData, () => {
  it('should render', () => {
    const component = renderer
      .create(
        <DrugListStructuredData
          drugNames={['First cool drug', 'Second cool drug']}
        />,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
