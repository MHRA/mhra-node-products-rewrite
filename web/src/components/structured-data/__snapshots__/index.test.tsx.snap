// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrugListStructuredData should render 1`] = `
<script
  dangerouslySetInnerHTML={
    Object {
      "__html": "{\\"@context\\":\\"https://schema.org\\",\\"@type\\":\\"ItemList\\",\\"itemListElement\\":[{\\"@type\\":\\"ListItem\\",\\"position\\":0,\\"item\\":{\\"@type\\":\\"Drug\\",\\"name\\":\\"First cool drug\\",\\"url\\":\\"https://products.mhra.gov.uk/product?product=First%20cool%20drug\\"}},{\\"@type\\":\\"ListItem\\",\\"position\\":1,\\"item\\":{\\"@type\\":\\"Drug\\",\\"name\\":\\"Second cool drug\\",\\"url\\":\\"https://products.mhra.gov.uk/product?product=Second%20cool%20drug\\"}}]}",
    }
  }
  type="application/ld+json"
/>
`;

exports[`DrugStructuredData should render 1`] = `
<script
  dangerouslySetInnerHTML={
    Object {
      "__html": "{\\"@context\\":\\"https://schema.org\\",\\"@type\\":\\"Drug\\",\\"name\\":\\"My cool drug\\"}",
    }
  }
  type="application/ld+json"
/>
`;

exports[`SubstanceListStructuredData should render 1`] = `
<script
  dangerouslySetInnerHTML={
    Object {
      "__html": "{\\"@context\\":\\"https://schema.org\\",\\"@type\\":\\"ItemList\\",\\"itemListElement\\":[{\\"@type\\":\\"ListItem\\",\\"position\\":0,\\"item\\":{\\"@type\\":\\"Substance\\",\\"name\\":\\"First cool substance\\",\\"url\\":\\"https://products.mhra.gov.uk/substance?substance=First%20cool%20substance\\"}},{\\"@type\\":\\"ListItem\\",\\"position\\":1,\\"item\\":{\\"@type\\":\\"Substance\\",\\"name\\":\\"Second cool substance\\",\\"url\\":\\"https://products.mhra.gov.uk/substance?substance=Second%20cool%20substance\\"}}]}",
    }
  }
  type="application/ld+json"
/>
`;

exports[`SubstanceStructuredData should render 1`] = `
<script
  dangerouslySetInnerHTML={
    Object {
      "__html": "{\\"@context\\":\\"https://schema.org\\",\\"@type\\":\\"Substance\\",\\"name\\":\\"My cool substance\\"}",
    }
  }
  type="application/ld+json"
/>
`;
